src/main/groovy/edu/oregonstate/mist/metaxe/XEAppDAO.groovy

```diff
@@ -0,0 +1,123 @@
+package edu.oregonstate.mist.metaxe
+
+import com.fasterxml.jackson.core.JsonParseException
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.JsonMappingException
+import javax.ws.rs.core.UriBuilder
+
+@groovy.transform.TypeChecked
+class XEAppDAO {
+    private URI esUrl
+    private ObjectMapper mapper = new ObjectMapper()
+
+    XEAppDAO(URI elasticsearchUrl) {
+        this.esUrl = elasticsearchUrl
+    }
+
+    ESResult getById(String id) {
+        def builder = UriBuilder.fromUri(this.esUrl)
+        def url = builder.path('{id}').build(id).toURL()
+
+        println "get by id: ${url}"
+
+        InputStream jsonStream
+        try {
+            jsonStream = url.newInputStream()
+        } catch (FileNotFoundException e) {
+            // 404 -> return null
```

Comment:
How about using the log object to put some debug or error messages for these use cases?